---
int8 end_result
---
float64 angle